jstreegrid.js

```diff
@@ -857,16 +934,30 @@
 			}
 			return result;
 		};
-		this._hide_grid = function (node) {
-			var children = node && node.children_d ? node.children_d : [], i;
-			// go through each column, remove all children with the correct ID name
-			findDataCell(this.uniq,children,this._gridSettings.gridcols).remove();
+    this._hide_grid = function (node) {
+      if (!node) { return true; }
+      this._detachColumns(node.id);
+      var children = node.children ? node.children : [], cols = this._gridSettings.columns || [], treecol = this._gridSettings.treecol;
+      // try to remove all children
+      for (var i = 0, len = children.length; i < len; i++) {
+        var child = this.get_node(children[i]);
+        // go through each column, remove all children with the correct ID name
+        for (var j = 0, lenj = cols.length; j < lenj; j++) {
+          if (j === treecol) { continue; }
+          findDataCell(this.uniq, child.id, j, $(this._domManipulation.columns[j])).remove();
+        }
+        if (child.state.opened) { this._hide_grid(child);}
+      }
+			
+      
+      this._reattachColumns(node.id);
 		};
 		this.holdingCells = {};
-		this.popHoldingCells = function (obj,col,hc) {
+    this.popHoldingCells = function (obj, col, hc) {
+      if (obj.state.hidden) { return $(); }
```

Comment:
Maybe you're right. I haven't got the holding-cells concept by now.
The drawing  procedures in the original control, don't have such a concept. Perhaps it would be better to handle the whole drwaing like in the tree and use nested unsigned lists instead of divs. The idea of nested elements is a very clean. especially when removing/hiding child elements!